src/psyclone/tests/psyir/nodes/loop_test.py

```diff
@@ -143,6 +166,55 @@ def test_loop_navigation_properties():
     assert loop.step_expr.value == "2"
 
 
+def test_loop_dag_name():
+    '''Test the dag_name method of Loop.'''
+    loop = make_loop()
+    assert loop.dag_name == "loop_1"
+    # Make the Loop an orphan to test the error handling.
+    loop.detach()
+    with pytest.raises(InternalError) as err:
+        _ = loop.dag_name
+    assert ("Cannot generate DAG name for loop node 'Loop[variable:'i']"
+            in str(err.value))
+
+
+def test_loop_node_str(monkeypatch):
+    '''Test the node_str method of Loop.'''
+    loop = make_loop()
+    assert loop.node_str(colour=False) == "Loop[variable='i']"
+
+    # Rather than mess about with colour codes, it's simpler if we
+    # monkeypatch the 'coloured_name' method (inherited from Node).
```

Comment:
Clever, I need to remember that :)

Reply:
Thanks. For some reason I had a lot of trouble getting the arguments right. (I tried to allow for `self` but that didn't work.)